pkgs/games/classicube/default.nix

```diff
@@ -0,0 +1,56 @@
+{ stdenv, fetchFromGitHub, curl, libX11, libGL, openal, dos2unix, liberation-sans-narrow, bash }:
+
+stdenv.mkDerivation rec {
+  pname = "ClassiCube";
+  version = "1.0.6";
+
+  src = fetchFromGitHub {
+    owner = "UnknownShadow200";
+    repo = pname;
+    rev = version;
+    sha256 = "1a0ycqixww965l6yk7q8n17d6laci0qpbjw4gf586711wcv4h4kr";
+  };
+
+  nativeBuildInputs = [ dos2unix ];
+
+  postUnpack = ''
+    sourceRoot=$sourceRoot/src
+  '';
+
+  prePatch = ''
+    dos2unix Platform.c
+  '';
+
+  patches = [ ./font-patch.patch ];
+
+  font = "${liberation-sans-narrow}/share/fonts/truetype/LiberationSansNarrow-Regular.ttf";
+
+  postPatch = ''
+    substituteInPlace Platform.c --replace %fontdir% ${font}
+    substituteInPlace Makefile --replace JOBS=1 JOBS=${ if enableParallelBuilding then "$NIX_BUILD_CORES" else "1"}
+  '';
+
+  buildInputs = [ curl libX11 libGL openal ];
+
+  #We need to copy the game in a writable folder, as it use it's installasion folder to save data
+  installPhase = ''
+    mkdir -p $out/bin
+    cp ClassiCube $out/bin/.ClassiCube
+    echo "#!${bash}/bin/bash
+      mkdir -p ~/.ClassiCube
+      rm -f ~/.ClassiCube/ClassiCubeBin
+      cp $out/bin/.ClassiCube ~/.ClassiCube/ClassiCubeBin
+      ~/.ClassiCube/ClassiCubeBin
+      rm -f ~/.ClassiCube/ClassiCubeBin" > $out/bin/ClassiCube
```

Comment:
I ran into the same situation when packaging clonehero: https://github.com/NixOS/nixpkgs/pull/67643/commits/5993343f84164d4838bffb569a6aa58e9cc63103#diff-4f8b4146c1be22ee9f7b8b81423c6534R87-R88.

I discovered that you can fake argv[0] so that the binary thinks it was launched from another location. That way you don't need to copy the binary.

I would also recommend following the XDG Base Directory Specification rather storing files directly in the user's home directory: https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html.

Reply:
Hmm, actually looking into the source code it looks like it reads from '/proc/self/exe' rather than argv[0]. I tested it with a sample script and the `exec -a` workaround doesn't seem to work.

Reply:
Now I found the function that indicate the function that find the binary name, I can just patch it (I suppose)

Reply:
Can you please patch it? This is a not very great workaround.